src/stateless/slider.vue

```diff
@@ -0,0 +1,546 @@
+                let position = (index - 1) / (this.ticksCount - 1) * this.sliderWidth
+                // threshold for ticks disappearing under labels
+                let minThreshold = this.$refs.min.clientWidth + this.labelPadding
+                let maxThreshold = this.sliderWidth - this.$refs.max.clientWidth - this.labelPadding
+                hidden = position <= minThreshold || position >= maxThreshold
+            }
+
+            return {
+                'active': index - 1 <= this.lastActiveIndex,
+                'hidden': hidden,
+            }
+        },
+    },
+}
+</script>
+
+<style lang="less" scoped>
+@import (reference) 'variables';
+
+.slider {
+    margin-bottom: 15px;
+    height: 60px;
+
+    &__row {
+        display: flex;
+    }
+
+    &__label {
+        height: 13px;
+        font-size: 11px;
+        color: @dolphin;
+        font-family: @regular-text-font;
+    }
+
+    &__input {
+        width: 50px;
+        margin-right: 10px;
+    }
+}
+
+.slider-helper-text {
+    height: 17px;
+}
+
+.ruler {
+    position: relative;
+    display: flex;
+    justify-content: space-between;
+    padding-top: 8px;
+
+    &__tick {
+        width: 1px;
+        height: 7px;
+        margin-top: 3px;
+        background-color: @gunpowder;
+        transition: background-color @form-element-transition-time ease-out;
+
+        &--hidden {
+            visibility: hidden;
+        }
+    }
+
+    &__label {
+        color: @dolphin;
+        font-size: 10px;
+        font-family: @regular-text-font;
+    }
+
+    &__ticks {
+        position: absolute;
+        width: 100%;
+        display: flex;
+        justify-content: space-between;
+    }
+}
+
+.rail {
+    height: 2px;
+    position: absolute;
+
+    &--active {
+        background-color: @royal-blue;
+        width: 0;
+    }
+
+    &--passive {
+        background-color: @gunpowder;
+        width: 100%;
+        transition: background-color @form-element-transition-time ease-out;
+    }
+
+    &--exceeds {
+        background: linear-gradient(90deg, @royal-blue 0%, @progress-blue 100%);
+    }
+
+    &__handle {
+        position: absolute;
+        bottom: -1px;
+        left: 0;
+        transform: translateX(-50%) translateY(50%);
+        width: 14px;
+        height: 14px;
+        box-sizing: border-box;
+        border-radius: 50%;
+        background-color: @very-light-gray;
+        border: 2px solid @extremely-dark-gray;
+        transition: transform @opening-animation-time-content ease-out;
+
+        &-dot {
+            width: 2px;
+            height: 2px;
+            border-radius: 50%;
+            background-color: @extremely-dark-gray;
+            position: absolute;
+            bottom: 0;
+            top: 0px;
+            right: 0px;
+            left: 0px;
+            margin: auto;
+            opacity: 0;
+            transition: opacity @opening-animation-time-content ease-out;
+        }
+    }
+}
+
+.bar {
```

Comment:
Shouldn't this be `slider__bar`? We usually have single BEM block per component (in this case `slider`). Same for `rail`, `ruler`, `slider-helper-text`.

Reply:
But how do we decide which elements should be blocks?`bar` is also a very generic class name so we might accidentally set some properties for it in another css file.

Reply:
I see what you mean. Other non-scoped styles could influence this rule by accident. To be safer I'd then use `slider-bar` for block.